### ISSUE TITLE
Change for Tutorial's finger15.py

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10408]
|<img alt="fresh's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| fresh reported|
|-|-|
|Trac ID|trac#531|
|Type|defect|
|Created|2004-03-03 21:21:48Z|

<details><summary>Searchable metadata</summary>

```
trac-id__531 531
type__defect defect
reporter__fresh fresh
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__documentation documentation
time__1078348908000000 1078348908000000
changetime__1472479315298882 1472479315298882
version__ 
owner__hypatia hypatia
cc__hypatia cc__fresh
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="fresh's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|fresh commented|
|-|-|

```
#!html
<pre>
The code for get request smelled bad to a number of people on #twisted, so
here's an alternate implementation:

    # we treat the path as the username
    def handleRequest(self, username, request ):
        motd = self.users.get(username)
        username = cgi.escape(username)
        if motd is not None:
            motd = cgi.escape(motd)
            text = '&lt;h1>%s&lt;/h1>&lt;p>%s&lt;/p>' % (username,motd)
        else:
            text = '&lt;h1>%s&lt;/h1>&lt;p>No such user&lt;/p>' % username
        return static.Data(text, 'text/html')
        
    def getResource(self):
        r = resource.Resource()        
        r.getChild = self.handleRequest
        return r
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement commented|
|-|-|

```
#!html
<pre>
Any chance you can attach a diff -u file, using the file box below?
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="fresh's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|fresh commented|
|-|-|

```
#!html
<pre>
urm :-S

It's probably easier just to open up the file for the tutorial and paste my
method in...

Which is diff -u so mandatory?
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement commented|
|-|-|

```
#!html
<pre>
It's not mandatory, it's just easier to check.

To apply your change, I will do this:

 1. paste your code in
 2. check that the indentation is right and that the various copy+pastes haven't
stuffed it up
 3. open the original file alongside to check exactly what the changes are,
because  I've been bitten at least once by someone making major stylistic
changes to an example in addition to their fix that they didn't tell me about
 4. Commit the change

To apply a diff -u change I do this:

 1. Eyeball the lines marked with + and - to check what the changes actually are
 2. Apply the patch
 3. Commit the change

Anyway, it appears that what you meant when you said "it's probably easier" is
"it's probably easier for the submitter and a little harder for the person
checking and committing the change", or at least when the person in question is me.

I commit the changes regardless though, if they're OK, so the "mandatory" tag
doesn't apply to this situation.

Another thing that would be handy would be some kind of summary of the content
of the discussion in IRC (one or two sentences or IRC quotes or something)
because I'm not on IRC.

Expect a commit on the weekend, since I'm going to have to take half an hour or
so to work out how the original works and why yours is better.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="fresh's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|fresh commented|
|-|-|

```
#!html
<pre>
Hmm, I should dig through my logs really ;-)

I couldn't figure out how the getResource method worked so I asked for help.

itamar and a coupla others kinda went "ew, that smells", so I thought I'd
re-implement it once I understood it.

My code is functionally identical, and just a little easier on the eye...

I don't work from CVS checkouts of twisted, so I can appreciate this is a pain
for you, especially as I'm working from the web site as well, not the files on disk.

I apologise for that, I'm nota stickler for immediate responses, my usual home
is Zope land where responses can be anything from a week to a year to never...

Thanks for your help!
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement commented|
|-|-|

```
#!html
<pre>
Made this change, closing the bug.

I made a change to your version to subclass Resource rather than re-writing one
of its methods. Have a look at
http://cvs.twistedmatrix.com/cvs/doc/howto/listings/finger/finger15.py?rev=1.2&amp;content-type=text/vnd.viewcvs-markup

One last comment on the diff -u thing: I discovered while trying to make this
change this rule: "if you can't submit a diff, please attach the entire changed
file".

Closing bug.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="fresh's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|fresh commented|
|-|-|

```
#!html
<pre>
What would the "entire changed file" have been in this case, given that I was
working off the website...
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement commented|
|-|-|

```
#!html
<pre>
You can click just below the code on any of the finger examples to download the
code for that example in a text file. In that case, it would have been a whole
new finger15.py.

I really did have a little trouble figuring out where your changes went, I'm
unfamiliar with the tutorial.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_9"></a>|@rodrigc commented|
|-|-|

In changeset 60f1fc2b21c601338fab2e3442dd0f976e90d2cf
```
#!CommitTicketReference repository="" revision="60f1fc2b21c601338fab2e3442dd0f976e90d2cf"
Merge pull request #531 from twisted/8799-rodrigc-reactorSpawnProcess

Author: rodrigc
Reviewer: derwolfe, markrwilliams
Fixes: #8799

Change 'str' to 'bytes' in deprecation message
```
